lib/Driver/Driver.cpp

```diff
@@ -703,6 +703,17 @@ Driver::buildCompilation(const ToolChain &TC,
   OI.CompilerMode = computeCompilerMode(*TranslatedArgList, Inputs, BatchMode);
   buildOutputInfo(TC, *TranslatedArgList, BatchMode, Inputs, OI);
 
+  // Note: Batch mode handling of serialized diagnostics requires that all
+  // batches get to run. (This behavior could be limited to only when serialized
```

Comment:
It would be valuable to explain a bit more about *why* all batches get to run, either here, or somewhere else with a pointer to here. It would save someone a lot of head-scratching someday.